sel4-ftrace-merge/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# set minimum cmake verision
+cmake_minimum_required(VERSION 3.0.0)
+
+# set project name
+project(ftrace-merge)
+
+# add executable
+add_executable (ftrace-merge ftrace-merge.c)
+
+# set compiler options
+set(FLAGS_WAR_AS_ERR -Wall -Wpedantic -Werror)
+# set(FLAGS_MEM_DBG -fsanitize=address -fsanitize=leak -fsanitize=bounds)
+
+# specify include dir
+target_include_directories(ftrace-merge PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+target_compile_options(ftrace-merge PRIVATE ${FLAGS_WAR_AS_ERR} ${FLAGS_MEM_DBG})
+target_link_options(ftrace-merge  PRIVATE ${FLAGS_WAR_AS_ERR} ${FLAGS_MEM_DBG})
```

Comment:
Treat compiler and linker options separately, don't mush them up (i.e. own variables for preprocessor, compiler and linker flags. By convention these are called CPPFLAGS, CFLAGS and LDFLAGS respectively.

Reply:
Fixed.